### PR TITLE
Support download files also.

### PR DESCRIPTION
**Pull request by mohsinenar**

### CODE REVIEW

File: agent/virustotal.py [hunk 1]
```diff
@@ -16,16 +15,15 @@ class VirusTotalApiError(Error):
...
-        message: Message containing the file to scan.
+        file_content: Message containing the file to scan.
         api_key : Key for the virustotal api.
     Returns:
         response: The response of the Virus Total public API.
     """
-    file = message.data["content"]
-    file_md5_hash = hashlib.md5(file)
+    file_md5_hash = hashlib.md5(file_content)
     hash_hexa = file_md5_hash.hexdigest()
```
Comment (BlueSquare1) [inline agent/virustotal.py:27]:
```suggestion
    file_hash = hashlib.md5(file_content).hexdigest()
```

**Review by BlueSquare1 — CHANGES_REQUESTED** [review]

File: agent/file.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    """
+    response = requests.get(file_url, timeout=DOWNLOAD_REQUEST_TIMEOUT)
+    if response.status_code == 200:
+        return response.content
+
+    return None
+
+
+def get_file_content(message: m.Message) -> bytes | None:
+    """Get the file info from a message.
```
Comment (ErebusZ) [inline agent/file.py:34]:
```suggestion
    """Get the file content from a message.
```

File: tests/file_test.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    content = file.get_file_content(message)
+
+    assert content == b"content"
+
+
+def testGetFileContent_whenContentIsEmpty_shouldTryToDownloadTheFile(
+    requests_mock: rq_mock.mocker.Mocker,
+) -> None:
+    """Test that the file is downloaded when the content is empty."""
```
Comment (ErebusZ) [inline tests/file_test.py:22]:
```suggestion
    """Test that the file is downloaded when the content url is provided."""
```

**Review by ErebusZ — CHANGES_REQUESTED** [review]

File: agent/file.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    """
+    content = message.data.get("content")
+    if content is not None and isinstance(content, bytes):
+        return content
+    content_url: str | None = message.data.get("content_url")
+    if content_url is not None:
+        file_content: bytes | None = _download_file(content_url)
+        if file_content is not None:
+            return file_content
+    return None
```
Comment (rabsondev) [inline agent/file.py:50]:
```suggestion
        return file_content
```

You're still going to return the file content whether it's `None` or not, so I don't see the need for a check.

Comment (rabsondev) [inline agent/file.py:50]:
```suggestion
        return file_content
```

You're still going to return the file content whether it's `None` or not, so I don't see the need for a check.

**Review by rabsondev — CHANGES_REQUESTED** [review]

File: agent/file.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
+"""Collection of functions to handle files."""
+import requests
+import tenacity
+from ostorlab.agent.message import message as m
+
+DOWNLOAD_REQUEST_TIMEOUT = 60
```
Comment (elyousfi5) [inline agent/file.py:6]: i think 60 is a lot for requests timeout !

Comment (3asm) [reply agent/file.py:6]: It is a safe value. Bigger has little impact.

File: agent/virus_total_agent.py [hunk 6]
```diff
@@ -44,8 +46,9 @@ def process(self, message: msg.Message) -> None:
             VirusTotalApiError: In case the Virus Total api encountered problems.
             NameError: In case the scans were not defined.
         """
-        if message.data.get("content") is not None:
-            response = virustotal.scan_file_from_message(message, self.api_key)
+        file_content = file.get_file_content(message)
```
Comment (3asm) [inline agent/virus_total_agent.py:49]: Unit test?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: agent/virus_total_agent.py (hunk 6 above)
Comment (mohsinenar) [reply agent/virus_total_agent.py:49]: `tests/file_test.py`!

File: agent/virustotal.py (hunk 1 above)
Comment (mohsinenar) [reply agent/virustotal.py:27]: @BlueSquare1 Please stick with the part of the code that is affected by the PR.

File: agent/virus_total_agent.py [hunk 7]
```diff
@@ -32,6 +33,7 @@ def __init__(
         """
         super().__init__(agent_definition, agent_settings)
         self.api_key = self.args.get("api_key")
+        self.whitelist_types = self.args.get("whitelist_types", [])
```
Comment (amine3) [inline agent/virus_total_agent.py:36]: The argument should be added to ostorlab.yaml

Comment (amine3) [inline agent/virus_total_agent.py:36]: Where is whitelist_types is used?

**Review by amine3 — CHANGES_REQUESTED** [review]

File: agent/virus_total_agent.py (hunk 7 above)
Comment (mohsinenar) [reply agent/virus_total_agent.py:36]: This is part of the other PR, just pushed by mistake.

**Review by rabsondev — APPROVED** [review]
lgtm

**Review by 3asm — APPROVED** [review]

**Pull request merged**